File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int>mp;
        vector<int>res;
        for(int i=0;i<nums.size();i++){
            mp[nums[i]]=i;
        }
        for(int i=0;i<nums.size();i++){
            int key=target-nums[i];
            if(mp.find(key)!=mp.end()&& mp[key] != i){
                return {i,mp[key]};
            }
        }
        return {};
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
from collections import Counter
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counter=Counter(nums)
        heap=[]
        for key,val in counter.items():
            heapq.heappush(heap,(-val,key))
        res=[]
        while k>0:
            val,key=heapq.heappop(heap)
            res.append(key)
            k-=1
        return res

        